File: scrimage-filters/src/main/scala/com/sksamuel/scrimage/filter/OldPhotoFilter.scala
package com.sksamuel.scrimage.filter

import java.awt.Graphics2D

import com.sksamuel.scrimage.{Filter, Image}
import thirdparty.misc.DaisyFilter
import thirdparty.romainguy.{BlendComposite, BlendingMode}

/** @author Stephen Samuel */
object OldPhotoFilter extends Filter {

  val film = Image.fromResource("/com/sksamuel/scrimage/filter/film1.jpg")

  def apply(image: Image) {

    val daisy = new DaisyFilter()
    val filtered = daisy.filter(image.awt)

    val g2 = image.awt.getGraphics.asInstanceOf[Graphics2D]
    g2.drawImage(filtered, 0, 0, null)
    g2.dispose()

    g2.setComposite(BlendComposite.getInstance(BlendingMode.INVERSE_COLOR_DODGE, 0.30f))
    g2.drawImage(film.scaleTo(image.width, image.height).awt, 0, 0, null)
    g2.dispose()
  }
}

File: scrimage-filters/src/main/java/com/sksamuel/scrimage/filter/OldPhotoFilter.java
package com.sksamuel.scrimage.filter;

import com.sksamuel.scrimage.ImmutableImage;
import com.sksamuel.scrimage.ScaleMethod;
import thirdparty.misc.DaisyFilter;
import thirdparty.romainguy.BlendComposite;
import thirdparty.romainguy.BlendingMode;

import java.awt.*;
import java.awt.image.BufferedImage;
import java.io.IOException;

public class OldPhotoFilter implements IntFilter {

   private static final ImmutableImage film;

   static {
      try {
         film = ImmutableImage.loader().fromResource("/com/sksamuel/scrimage/filter/film1.jpg");
      } catch (IOException e) {
         throw new RuntimeException(e);
      }
   }

   @Override
   public void apply(ImmutableImage image) throws IOException {

      DaisyFilter daisy = new DaisyFilter();
      BufferedImage filtered = daisy.filter(image.awt());

      Graphics2D g2 = (Graphics2D) image.awt().getGraphics();
      g2.drawImage(filtered, 0, 0, null);

      BufferedImage filmSized = film.scaleTo(image.width, image.height, ScaleMethod.Bicubic).awt();
      BufferedImage filmSizedSameType = ImmutableImage.fromAwt(filmSized, image.awt().getType()).awt();

      g2.setComposite(BlendComposite.getInstance(BlendingMode.INVERSE_COLOR_DODGE, 0.30f));
      g2.drawImage(filmSizedSameType, 0, 0, null);
      g2.dispose();
   }
}

